accepted/wasmtime-one-dot-oh.md

```diff
@@ -0,0 +1,558 @@
+  proposal is highly likely to alter the public API of Wasmtime in some way
+  that is breaking, so such a proposal for string API-stability guarantees would
+  need at least some concession for when to do breaking changes. The most likely
+  form of this would be to batch up breaking changes, perhaps in an LTS-style
+  cadence. This means, though, that new features are artificially delayed in
+  Wasmtime and cannot land when they are originally implemented. Furthermore
+  there would be no way to land intermediate work which would continue to be
+  developed in-tree. Overall it's expected that users who want API-stability
+  from Wasmtime are sufficiently serviced with Wasmtime's LTS versions which
+  guarantee API-stability but lack new features. At this time an alternative of
+  API-stable releases still getting new non-API-breaking features is seen as too
+  much of a hindrance to the development of Wasmtime itself.
+
+* Technically there's nothing really stopping feature-development being
+  backported to LTS versions so long as the feature doesn't change any APIs.
+  This proposal, however, only indicates that security issues and bugs are
+  backported to LTS versions. This is intended to be a relatively conservative
+  starting position where we could still backport features as necessary if
+  someone's willing to put in the work. The fear, though, is that the backport
+  process becomes relatively complicated and LTS backports are likely less
+  battle-tested than changes on `main` due to the nature of mismatch between the
+  original state of `main` and the LTS itself. To help make what is already a
+  somewhat complicated process a bit simpler, this proposal states that new
+  features are not backported to LTS releases.
+
+* LTS versions in theory could be dropped entirely. There aren't known users who
+  are specifically asking for LTS versions and existing production users of
+  Wasmtime are likely to stay with released versions to get improvements as they
+  come out. This proposal assumes, however, that there are developers we're not
+  actively hearing from who would benefit from such LTS versions and we'd like
+  to accomodate them. If in practice it turns out that there is very little
+  usage of LTS versions we can consider dropping support for them at a later
+  date when such a conclusion is made (probably another RFC).
+
+* As currently proposed Wasmtime is guaranteed to receive a new major version
+  every 4 weeks. It might be the case, though, that within those 4 weeks of
+  development that Wasmtime didn't actually have any API-breaking changes occur.
+  Theoretically this could allow for a minor release of the form 2.1.0, for
+  example, if 2.0.0 was the previous release. There are a number of downsides
+  with trying to do this, though, including:
+
+  * Wasmtime developers now have to keep track of breaking changes and determine
+    whether any landed in the 4-week release window.
```

Comment:
Yes! I feel the project _should_ keep track of breaking changes; the user will have a lot harder time figuring that out (see my semver comment above).

Reply:
This might be too strong a statement as written: as discussed elsewhere in the explainer, the idea is very much to be mindful of API stability and not make updating harder than it needs to be. So actually breaking changes should be documented as such, which means that this sentence should still be true as per this plan.

Reply:
Yeah I'm not getting across what I was thinking with this bullet. It's moreso that having an unpredictable versioning scheme makes it harder for releases, automation, and such. Having to programmatically detect a breaking change somehow does not seem super viable and could complicate what's already bound to be a pretty complicated set of automation for these releases.

I consider it a bare minimum that at all times we're keeping track of breaking changes and logging them in a changelog of some sort.